### PR TITLE
[lexical-playground] 3 Bug Fixes, 1 UX Improvement: All Regarding Excalidraw Node

### PR DESCRIPTION
**Pull request by neysanfoo**
#### Steps to Reproduce
1. Create some drawing (non-empty)
2. Save the drawing
3. Edit the drawing
4. Clear all the elements
5. Discard Changes

Current Behaviour:
- Drawing gets deleted due to bug in "discard" function that is deleting the node if scene is clear.

New Behaviour:
- Just close the modal. So the changes are actually discarded.

#### Steps to Reproduce
1. Create some drawing (non-empty)
2. Discard Changes

Current Behaviour:
- An excalidraw node is still inserted.

New Behaviour:
- Doesn't insert a node.

### Video of Bug 1 and 2 (Before Changes)
https://github.com/user-attachments/assets/c9379c0b-28ce-4386-9f8a-5827f33fe019

### Video of Bug 1 and 2 (After Changes)
https://github.com/user-attachments/assets/2c349e3a-14cb-4389-b82e-eb83e6ac679c

#### Steps to Reproduce
1. Create some drawing (non-empty)
2. Save the drawing
3. Undo x2
4. Redo

Current Behaviour:
- Firstly, it takes 2 undos to remove the node
- Modal opens up again on first redo

New Behaviour
- Do not add the intermediate steps, i.e. opening the modal, saving changes, or discarding changes, to the undo stack as per #4283 

### Video of UX Improvement 1 (Before Changes)

https://github.com/user-attachments/assets/be581e7c-ecac-40c7-a09a-34ed7601afd9

### Video of UX Improvement 1 (After Changes)

https://github.com/user-attachments/assets/21fe4cd8-2da0-4033-b938-40d3c2a669cc

### Bug 3
Lastly, removed a small bug DecoratorNode doesn't have an exportJSON method.

### Future Work
- Track the history of resizing, similar to how the image component works

### CODE REVIEW

File: packages/lexical-playground/src/nodes/ExcalidrawNode/index.tsx [hunk 1]
```diff
@@ -84,7 +84,6 @@ export class ExcalidrawNode extends DecoratorNode<JSX.Element> {
 
   exportJSON(): SerializedExcalidrawNode {
     return {
-      ...super.exportJSON(),
```
Comment (etrepum) [inline packages/lexical-playground/src/nodes/ExcalidrawNode/index.tsx:87]: Nice catch!

**Review by etrepum — COMMENTED** [review]
It's not really clear why the history-merge tag was added to these updates, can you explain? I think that might not be the desired behavior.

**Comment by neysanfoo** [comment]
> It's not really clear why the history-merge tag was added to these updates, can you explain? I think that might not be the desired behavior.

So the history-merge tags that were added are to prevent having the modal in the undo stack (ref to videos of UX improvement). The rest of the code changes are to handle the bugs.

Consider the case: Open modal -> Draw -> Save. I guess there are 2 options for how I can imagine the UX working for undoing and redoing, the first is to count the entire process of opening, drawing and saving as a single action, which is what im proposing. The second option would be 1 undo opens the modal with the current data loaded and then another undo removes the node altogether, and similarly a single redo will open the modal with the data loaded and then another redo will put the image back.

However the current process right now is one undo will delete the image without opening the modal, the second undo will actually delete the node and then a redo will show an empty modal with no data loaded in it, which is a bit weird because we didnt see the modal open when undoing, and also its empty.

So to summarize the history-merge tags were added to the 3 functions to treat the actions of open modal -> draw -> save, open modal -> draw/dont draw -> discard, edit image -> change/no change -> save, edit image -> change/no change -> discard as single actions.

**Comment by etrepum** [comment]
I am not particularly familiar with this code but I suspect that this might cause some problems with collab which is the only reason I can think of to have the node's state update while the modal is open. In the single user case I think a better strategy would be to only update the state that the decorator exposes to the editor when the modal closes?

**Comment by neysanfoo** [comment]
I just tried excalidraw using collab. It doesn't seem to be even usable because modal opens on both screens. Would having the toolbar open the excalidraw modal, and then only creating the node on save make sense? I think that would solve both problems.

**Comment by etrepum** [comment]
I think so, at least that's the approach I would take if I was building that node + plug-in

**Comment by neysanfoo** [comment]
I have moved the code for opening the modal into the ui folder and decoupled the creation of the node from the opening/editing of the modal. The node is only created on saving the modal.

**Comment by neysanfoo** [comment]
Any advice how to fix the failing tests?

**Comment by ivailop7** [comment]
It was the flaky set of tests, all good now. @etrepum happy to merge?

**Review by etrepum — APPROVED** [review]
I tested this out and it the resizing of these nodes does not persist in history, although that is not a regression in this particular PR (maybe the last one though?)

**Comment by neysanfoo** [comment]
> I tested this out and it the resizing of these nodes does not persist in history, although that is not a regression in this particular PR (maybe the last one though?)

I think this has never worked, I'm planning on fixing this next.
